torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -3063,8 +3065,9 @@ def _summon_full_params(
 
         self._clear_grads_if_needed()
         free_unsharded_flat_params = [handle.needs_unshard() for handle in self._handles]
-        self._unshard(self._handles)
-        self._streams["computation"].wait_stream(self._streams["unshard"])
+        # No need to call `wait_stream()` since we unshard in the computation
```

Comment:
Curious, why would we want to move this to computation stream?

Reply:
This allows us to use caching allocator blocks from the computation stream for these all-gathers, which should help avoid over-allocating blocks to the unshard stream.